### ISSUE TITLE
Issue with build_tiles with transit

### ISSUE THREAD

**Issue by hn0017** [valhalla/valhalla#4238]
I am using valhalla_build_tiles with transit data but having issue in the process. The process is aborting and the error is as follows:

2023/08/04 07:53:22.170596 [INFO] Adding 12 transit tiles to the local graph...
2023/08/04 07:53:22.277542 [WARN] Could not find connection point for in/egress near: 28.989537,77.700953
terminate called after throwing an instance of 'std::bad_alloc'
  what():  std::bad_alloc
Aborted (core dumped)

I am new to valhalla but on inspecting found something.
If I am not wrong this may be the issue in transitbuilder.cc file

    if (!closest_edge) {
      LOG_WARN("Could not find connection point for in/egress near: " +
               std::to_string(egress_ll.second) + "," + std::to_string(egress_ll.first));
    }

    // TODO: if the point we found is further away than the tile edge then there could be a better
    //  match in an adjacent tile. For now we should detect this and log, to get a feeling of
    //  prevalence. In the future we should look at other tiles (once we fix the below issue this will
    //  be easy)

    // flip the shape if we have to
    std::vector<PointLL> edge_shape = closest_edgeinfo->shape();
    if (!closest_edge->forward()) {
 When changed to this , build_tiles completed sucessfully

    if (!closest_edge) {
       LOG_WARN("Could not find connection point for in/egress near: " +
               std::to_string(egress_ll.second) + "," + std::to_string(egress_ll.first));
       continue;
    }

Just wanted to know that am I looking in the right direction

**Comment by pfennig42** [comment]
I second this problem. It happens, when there is a point in the GTFS-Data, that is not in the .pbf files. Throwing a warning instead of an error should be more appropriate

**Comment by nilsnolde** [comment]
Yeah, that's still a TODO on our list for transit stuff. It has actually fairly high priority and I'd like to get that squashed soon-ish.

> When changed to this , build_tiles completed sucessfully
if (!closest_edge) {
   LOG_WARN("Could not find connection point for in/egress near: " +
           std::to_string(egress_ll.second) + "," + std::to_string(egress_ll.first));
   continue;
}

@hn0017 it's better than a segfault for sure:) I think the result will be that you won't be able to use that station/platform as origin/destination transit stop, as OSM didn't connect. IIRC it can still be used for in-station transfers. 

In fact, I don't know if we shouldn't just do what you're doing (at least in spirit, didn't look at the code or all the consequences). If a station is outside the bounds of OSM, we could also choose to remove that platform/station from the tiles so it wouldn't be usable anymore by the router. However, if we can still use it for transit transfers it's worth keeping and just not connect it to OSM.

WDYT @kevinkreiser ?

**Comment by hn0017** [comment]
I think here if  'closest_edge ' is null it logs an error but goes on to the next line which is ' if (!closest_edge->forward()) '  and this might be the one causing segmentation fault thats why I thought to skip this iteration further by using a continue.

**Comment by hn0017** [comment]
I think here if 'closest_edge ' is null it logs an error but goes on to the next line which is ' if (!closest_edge->forward()) ' and this might be the one causing segmentation fault thats why I thought to skip this iteration further by using a continue.

**Comment by nilsnolde** [comment]
Jep, I can see that. I'm just adding that we could also choose to remove the station/platform entirely from the graph at that point (they're added to the graph way earlier). And questioning if that'd be the right thing to do.. We should do a gurka test with that scenario and see if such OSM-unconnected (but not deleted) stations/platforms can still be used for transfer (which I'm fairly sure they can be, at least in-station transfers).

**Comment by xxhh1112** [comment]
t

**Comment by johnnydependents** [comment]
Is there an update to this? I keep running into it and can't make any progress

**Comment by K4ST0R** [comment]
Got the same issue, is there any update on this ?
